bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java

```diff
@@ -136,6 +172,40 @@ protected void initModel() {
                             title = damTitle;
                         }
                     }
+
+                    //check "Enable DM features" checkbox
+                    //check DM asset - check for "dam:scene7File" metadata value
+                    String dmAssetName = asset.getMetadataValue(Scene7Constants.PN_S7_FILE);
+                    if(isDmFeaturesEnabled && (!StringUtils.isEmpty(dmAssetName))){
+                        //image is DM
+                        dmImage = true;
+                        //check for publish side
+                        boolean isWCMDisabled =  (com.day.cq.wcm.api.WCMMode.fromRequest(request) == com.day.cq.wcm.api.WCMMode.DISABLED);
+                        try {
+                            String dmServerUrl;
+                            if (!isWCMDisabled) {
+                                //for Author
+                                dmServerUrl = "/is/image/";
+                                String[] productionImageUrls = publishUtils.externalizeImageDeliveryAsset(assetResource);
+                                try {
+                                    ModifiableValueMap map = resource.adaptTo(ModifiableValueMap.class);
+                                    if (map != null) {
+                                        map.put(PN_IMAGE_SERVER_URL, productionImageUrls[0] + "/is/image/");
+                                        resource.getResourceResolver().commit();
```

Comment:
Please remove the write operation inside of the SlingModel. Possible replacement would be a SlingPostProcessor which is writing to the repository only when the asset resource has changed and DM is enabled.